lib/internal/fs/utils.js

```diff
@@ -698,7 +699,13 @@ const getValidMode = hideStackFrames((mode, type) => {
 });
 
 const validateStringAfterArrayBufferView = hideStackFrames((buffer, name) => {
-  if (typeof buffer !== 'string') {
+  if (
+    typeof buffer !== 'string'
+    || !buffer
+    || typeof buffer !== 'object'
+    || !ObjectPrototypeHasOwnProperty(buffer, 'toString')
+    || typeof buffer.toString !== 'function'
+  ) {
```

Comment:
This doesn't seem like the correct logic. These additional checks would now be executed when `typeof buffer === 'string'`, which aside from unnecessary, could especially cause issues for empty strings with the `!buffer` check for example.

Reply:
ah thanks, let me fix that :-)